### ISSUE TITLE
LocalSettings throws an UnauthorizedAccessException after AppInstance.Restart

### ISSUE THREAD

**Issue by sjb-sjb** [microsoft/WindowsAppSDK#3957]
### Describe the bug

LocalSettings throws an UnauthorizedAccessException after AppInstance.Restart.
How can I fix this? 

Expected: should be able to access LocalSettings after a restart. 

### Steps to reproduce the bug

Using Windows 10 and Windows App SDK, create a WinUI 3 desktop application using the template. 

Change MainWindow.xaml to contain this: 

    <StackPanel Orientation="Vertical" HorizontalAlignment="Center" VerticalAlignment="Center">
        <TextBlock x:Name="TimeTextBlock">(unkown)</TextBlock>
        <Button x:Name="SetButton" Click="Set_Click">Set time</Button>
        <Button x:Name="GetButton" Click="Get_Click">Get time</Button>
        <Button x:Name="RestartButton" Click="Restart_Click">Restart</Button>
        <TextBlock x:Name="StatusTextBlock">Startup</TextBlock>
    </StackPanel>

Change MainWindow.xaml.cs to contain this: 

        public MainWindow()
        {
            this.InitializeComponent();
        }

        private Windows.Storage.ApplicationDataContainer LocalSettings => this._localSettings ??= Windows.Storage.ApplicationData.Current.LocalSettings;
        private Windows.Storage.ApplicationDataContainer? _localSettings;

        private void Set_Click(object sender, RoutedEventArgs args)
        {
            try {
                DateTimeOffset now = DateTimeOffset.Now;
                this.LocalSettings.Values["Time"] = now;
                this.StatusTextBlock.Text = "Set value " + now.ToString();
            } catch (Exception e) {
                this.StatusTextBlock.Text = $"Exception when setting value, type '{e.GetType().Name}' message '{e.Message}'.";
            }
        }

        private void Get_Click(object sender, RoutedEventArgs args)
        {
            try {
                DateTimeOffset? time;
                if (this.LocalSettings.Values.TryGetValue("Time", out object? value)) {
                    time = (DateTimeOffset)value;
                } else {
                    time = null;
                }
                this.TimeTextBlock.Text = time?.ToString() ?? "(null)";
                this.StatusTextBlock.Text = "Got value shown above from LocalSettings";
            } catch (Exception e) {
                this.StatusTextBlock.Text = $"Exception when getting value, type '{e.GetType().Name}' message '{e.Message}'.";
            }
        }

        private void Restart_Click(object sender, RoutedEventArgs e)
        {
            AppRestartFailureReason failure = Microsoft.Windows.AppLifecycle.AppInstance.Restart("");
            this.StatusTextBlock.Text = $"Restart failed, '{failure.ToString()}'";
        }

Run the app. Click on the Get and Set buttons, observe it works. 
Then click on Restart. After the application restart, click on Get or Set and receive an UnauthorizedAccessException.

### Expected behavior

Should be able to access LocalSettings after a restart.

### NuGet package version

None

### Packaging type

Packaged (MSIX)

### Windows version

Windows 10 version 22H2 (19045, 2022 Update)

### IDE

Visual Studio 2022

**Comment by sjb-sjb** [comment]
Hi @DarranRowe I am using C# and the packaged desktop (WinUI 3) app template. 
Not sure where the partial trust setting is but if you tell me I will take a look.

**Comment by DarranRowe** [comment]
It is very unlikely that you have it set then. The default isn't partial trust, and it is a setting that you have to go out of your way to enable.
On the off chance that you enabled it though. If you have a separate packaging project, then you have to select the project name in solution exporer and then select it in properties.

![Screenshot 2023-11-16 192224](https://github.com/microsoft/WindowsAppSDK/assets/52577874/b1d7c301-a819-495d-8fef-9d0333f900c9)

If you are using the single project layout, then it is even less likely that you could stumble onto this. You would have to manually edit the Package.appxmanifest file and type out either windows.partialTrustApplication and/or appContainer.

**Comment by sjb-sjb** [comment]
Yes it’s the single project layout and I did not type the phrase you mention.

**Comment by mominshaikhdevs** [comment]
related issues: 

1. https://github.com/microsoft/WindowsAppSDK/issues/2621.

2. https://github.com/microsoft/WindowsAppSDK/issues/5167.

3. https://github.com/microsoft/WindowsAppSDK/issues/5387.
